kotlin/src/test/TestApiSettings.kt

```diff
@@ -15,18 +43,25 @@ val quotedMinimum = """
 base_url='https://my.looker.com:19999'
 """.trimIndent()
 
+val withEnvPrefix = """
+[Looker]
+base_url=https://my.looker.com:19999
+environmentPrefix=$mySDK
+""".trimIndent()
+
 val mockId = "IdOverride"
 val mockSecret = "SecretOverride"
 
 class MockSettings(contents: String) : ConfigurationProvider by ApiSettings.fromIniText(contents) {
```

Comment:
Alternately, this might be a bit easier to understand, though you'll have to change how get a MockSettings in your tests to use the new factory.

```
class MockSettings(
    private val settings: ConfigurationProvider
) : ConfigurationProvider by settings {
    companion object {
        fun fromIniText(contents: String): ConfigurationProvider {
            return MockSettings(ApiSettings.fromIniText(contents))
        }
    }

    override fun readConfig(): Map<String, String> {
        return settings.readConfig().plus(
            mapOf(
                ....
            )
        )
    }
}
```

This is basically what I was trying to express earlier about how to implement SdkSettings, before we discovered we could just drop that in favor of moving ApiSettings.

Reply:
I'll fiddle with that some. Thanks for the ideas.

Reply:
There's a slightly better flow/example in the ApiSettings tests now.